frontend/src/metabase-lib/v1/queries/utils/query-time.js

```diff
@@ -247,25 +247,9 @@ export function getStartingFrom(mbql) {
   return [num, unit];
 }
 
-export function formatStartingFrom(bucketing, n) {
-  const suffix = n >= 0 ? "from now" : "ago";
-  switch (bucketing) {
-    case "minute":
-      return ngettext(msgid`minute ${suffix}`, `minutes ${suffix}`, n);
-    case "hour":
-      return ngettext(msgid`hour ${suffix}`, `hours ${suffix}`, n);
-    case "day":
-      return ngettext(msgid`day ${suffix}`, `days ${suffix}`, n);
-    case "week":
-      return ngettext(msgid`week ${suffix}`, `weeks ${suffix}`, n);
-    case "month":
-      return ngettext(msgid`month ${suffix}`, `months ${suffix}`, n);
-    case "quarter":
-      return ngettext(msgid`quarter ${suffix}`, `quarters ${suffix}`, n);
-    case "year":
-      return ngettext(msgid`year ${suffix}`, `years ${suffix}`, n);
-  }
-  return "";
+export function formatStartingFrom(unit, n) {
+  const unitText = Lib.describeTemporalUnit(unit, n).toLowerCase();
+  return n >= 0 ? t`${unitText} from now` : t`${unitText} ago`;
```

Comment:
This is unrelated to this PR (and I've approved it), but in general [it's bad practice](https://localization.blog/2022/05/16/i18n-best-practices-keep-it-together/) to break up temporal strings and then manually stitch them back together like this.

This because it make make it harder for translators to surmise the context of the string, and it might make (correct) translation to certain languages completely impossible.

I'm surprised this hasn't bitten us yet, but maybe we're just lucky with the languages we currently support.

Reply:
Actually, looking a bit deeper at this, why don't we use [`describe-relative-datetime`](https://github.com/metabase/metabase/blob/master/src/metabase/lib/temporal_bucket.cljc#L83) for this?

Reply:
It should be `year from now` and not `1 year from now`.

Reply:
I fully agree with [this comment](https://github.com/metabase/metabase/pull/49241/files#r1820229930). Though we'd need more MLv2 functions just for translations, which is not great. We probably should figure out a different approach eventually.

Reply:
Let's not block on this here.